### ISSUE TITLE
user-defined loss function

### ISSUE THREAD

**Issue by supernan** [dmlc/xgboost#1235]
Can xgboost support user defined loss function? If it can do that which api can i use

Thank you~

**Comment by tqchen** [comment]
see 
- https://github.com/dmlc/xgboost/blob/master/demo/guide-python/custom_objective.py
- https://github.com/dmlc/xgboost/blob/master/R-package/demo/custom_objective.R
